src/Adapter/MSTest.CoreAdapter/Discovery/UnitTestDiscoverer.cs

```diff
@@ -80,15 +80,29 @@ internal UnitTestDiscoverer()
 
         internal void SendTestCases(string source, IEnumerable<UnitTestElement> testElements, ITestCaseDiscoverySink discoverySink)
         {
-            object navigationSession = null;
+            Dictionary<string, object> navigationSessions = new Dictionary<string, object>();
```

Comment:
Am loving this. 👍 Performance is one of our key pillars. Although we haven't heard much in terms of performance issues in mstest v2, we learnt the hard way that this could hit us hard down the line.